File: app/src/main/java/com/example/degoogle/data/RoomPackagesDatabase.java
package com.example.degoogle.data;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.example.degoogle.data.dao.PackagesDao;
import com.example.degoogle.data.entities.InstalledPackages;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

@Database(entities = {InstalledPackages.class}, version = 1, exportSchema = false)
public abstract class RoomPackagesDatabase extends RoomDatabase {

    public abstract PackagesDao packagesDao();

    private static volatile RoomPackagesDatabase INSTANCE;
    private static final int NUMBER_OF_THREADS = 4;
    public static final ExecutorService databaseWriteExecutor =
            Executors.newFixedThreadPool(NUMBER_OF_THREADS);

    public static RoomPackagesDatabase getDatabase(final Context context) {
        if (INSTANCE == null) {
            synchronized (RoomPackagesDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            RoomPackagesDatabase.class, "packages_database")
                            .build();
                }
            }
        }
        return INSTANCE;
    }
}

File: app/src/main/java/com/example/degoogle/data/RoomPackagesDatabase.kt
package com.example.degoogle.data

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.example.degoogle.data.entities.InstalledPackages

@Database(entities = [InstalledPackages::class], version = 1, exportSchema = false)
public abstract class RoomPackagesDatabase : RoomDatabase() {

    abstract fun packagesDao(): PackagesDao

        companion object {
            @Volatile
            private var INSTANCE: RoomPackagesDatabase? = null

            fun getDatabase(context: Context): RoomPackagesDatabase {
                // if the INSTANCE is not null, then return it,
                // if it is, then create the database
                return INSTANCE ?: synchronized(this) {
                    val instance = Room.databaseBuilder(
                        context.applicationContext,
                        RoomPackagesDatabase::class.java,
                        "item_database"
                    )
                        // Wipes and rebuilds instead of migrating if no Migration object.
                        // Migration is not part of this codelab.
                        .fallbackToDestructiveMigration()
                        .build()
                    INSTANCE = instance
                    // return instance
                   return instance
                }
            }


        }
}